IdfaIOS/CustomIdfa.m

```diff
@@ -0,0 +1,113 @@
+#import <AdSupport/AdSupport.h>
+#import <AppTrackingTransparency/AppTrackingTransparency.h>
+#import <CoreLocation/CoreLocation.h>
+
+//Please ensure this path points to the file 'Assets/Plugins/iOS/Amplitude/Amplitude.h'
+#import "../../Plugins/iOS/Amplitude/Amplitude.h"
+
+typedef NSString* (^AMPAdSupportBlock)(void);
+typedef NSDictionary* (^AMPLocationInfoBlock)(void);
+
+// Converts C style string to NSString
+// Note, this will produce C++ compiler/linker errors if this function is renamed to ToNSString or any function in Objective C++ files
+NSString* ToNSString2(const char* string)
+{
+    if (string)
+        return [NSString stringWithUTF8String: string];
+    else
+        return nil;
+}
+
+void setIdfaBlockInternal(const char* instanceName)
+{
+    NSLog(@"Setting idfa block");
+    AMPAdSupportBlock adSupportBlock = ^{
+        NSUUID *adUuid = [[ASIdentifierManager sharedManager] advertisingIdentifier];
+        NSString *result = [adUuid UUIDString];
+        return result;
+    };
+    [[Amplitude instanceWithName:ToNSString2(instanceName)] setAdSupportBlock:adSupportBlock];
+}
+
+@interface MyLocationManager : NSObject<CLLocationManagerDelegate> 
```

Comment:
I am also thinking maybe we don't even need to provide the actual implementation for them because they might need to fetch location with different settings.

Reply:
I chose to provide the location implementation as it is primarily written in Objective C. It was also really difficult for me to implement the whole series of callbacks. The user of course is free to change the implementation or ask for different permission/info as needed.